addons/mail/wizard/mail_compose_message.py

```diff
@@ -203,6 +203,40 @@ def _check_res_domain(self):
         for composer in self:
             composer._evaluate_res_domain()
 
+    @api.depends('composition_mode', 'model', 'parent_id', 'res_ids',
+                 'template_id')
+    def _compute_subject(self):
+        """ Computation is coming either form template, either from context.
+        When having a template with a value set, copy it (in batch mode) or
+        render it (in monorecord comment mode) on the composer. Otherwise
+        it comes from the parent (if set), or computed based on the generic
+        '_message_compute_subject' method in monorecord comment mode, or
+        set to False. When removing the template, reset it. """
+        for composer in self:
+            if composer.template_id:
+                composer._set_value_from_template('subject', force_void=False)
+            if (not composer.template_id or not composer.subject) and composer.composition_mode == 'mass_mail':
```

Comment:
Could be elif and not check subject to simplify?

In other cases you don't set anything when the computed value is "Falsy", so I was guessing it's not necessary anymore.

Actually, would rework a bit the logic:
- set temporary variable subject to False
- if template, try to set the value from template
- if subject is Falsy, use parent_id stuff
- assign subject to composer